subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/util/DefaultJavaForkOptionsTest.groovy

```diff
@@ -769,6 +829,36 @@ class DefaultJavaForkOptionsTest extends Specification {
         !options.isCompatibleWith(other2)
     }
 
+    def "cannot determine compatibility with jvmArgumentProviders"() {
+        def other = new DefaultJavaForkOptions(resolver)
+        def argumentProvider = new CommandLineArgumentProvider() {
+            @Override
+            Iterable<String> asArguments() {
+                return ['argFromProvider']
+            }
+        }
+        if (currentHasProviders) {
+            options.jvmArgumentProviders << argumentProvider
+        }
+        if (otherHasProviders) {
+            other.jvmArgumentProviders << argumentProvider
+        }
+
+        when:
+        options.isCompatibleWith(other)
+
+        then:
+        def thrown = thrown(UnsupportedOperationException)
+        thrown.message.contains('Cannot compare options which have jvmArgumentProviders configured.')
```

Comment:
`which have` -> `with`, like in test title?

Reply:
Done.